pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -972,39 +973,50 @@ func createCheckpointIfNotExists(
 	return nil
 }
 
+// RedactURIForErrorMessage redacts any storage secrets before returning a URI which is safe to
+// return to the client in an error message.
+func RedactURIForErrorMessage(uri string) string {
+	redactedURI, err := cloudimpl.SanitizeExternalStorageURI(uri, []string{})
+	if err != nil {
+		return ""
```

Comment:
Should we could return something like `<uri_failed_to_redact>` to have something here to avoid potentially confusing error messages?

Reply:
we _could_ also fail open instead and return the input on the basis that if it isn't actually a URI, then there's no secret params to redact. Downside would be a malformed URI that _does_ contain secrets is left unreacted... but there's only so much we can do. Keep in mind a SQL parse error sends you back the unparsable string, so we can only really promise to redact things we can make enough sense of to do so.

Reply:
done.

Reply:
oops didn't read the latest comment - I don't really have a strong preference here but I feel like we would raise fewer eyebrows by erring on the side of not showing anything if the URI is malformed. I realize this makes debugging any deformities in the actual URI harder but there is usually enough surrounding text to know where to look? Further, Paul's recommended sentinel would indicate that this is definitely a parsing error.

I could go either way if you think an open fail is better.

Reply:
I dunno. On the one hand, yeah, we've heard loud and clear from users who don't like seeing their secrets repeated back to them. On the other, if we get complete gibberish, we can't tell you what is a secret and what isn't, so "never echo a secret" isn't possible anyway -- if that secret is in a malformed SQL statement, it's getting echo'ed in the parse error before any of our code gets involved to do potential redaction. So I could see unparsable URI being in the same category of unparsable command in general, where we're no on the hook for redacting? 

OTOH, if we got this far, it's highly likely that it is a cloud URI and those _probably_ do contain secrets, so fail closed as _pretty good_ false positive rate. I dunno.